src/safeds/ml/nn/_model.py

```diff
@@ -0,0 +1,250 @@
+import copy
+from typing import Self
+
+import numpy as np
+import torch
+from torch import nn
+
+from safeds.data.tabular.containers import Column, Table, TaggedTable
+from safeds.exceptions import ClosedBound, ModelNotFittedError, OutOfBoundsError
+from safeds.ml.nn._fnn_layer import FNNLayer
+
+
+class RegressionNeuralNetwork:
+    def __init__(self, layers: list):
+        self._model = _PytorchModel(layers, is_for_classification=False)
+        self._batch_size = 1
+        self._is_fitted = False
+
+    def fit(self, train_data: TaggedTable, epoch_size: int = 25, batch_size: int = 1) -> Self:
+        """
+        Train the neural network with given training data.
+
+        The original model is not modified.
+
+        Parameters
+        ----------
+        train_data : TaggedTable
+            The data the network should be trained on.
+        epoch_size : int
+            The number of times the training cycle should be done
+        batch_size : int
+            The size of data batches that should be loaded at one time.
+
+        Raises
+        ------
+        ValueError
+            If epoch_size < 1
+            If batch_size < 1
+
+        Returns
+        -------
+        RegressionNeuralNetwork
+            The trained Model
+        """
+        if epoch_size < 1:
+            raise OutOfBoundsError(actual=epoch_size, name="epoch_size", lower_bound=ClosedBound(1))
+        if batch_size < 1:
+            raise OutOfBoundsError(actual=batch_size, name="batch_size", lower_bound=ClosedBound(1))
+        copied_model = copy.deepcopy(self)
+        copied_model._batch_size = batch_size
+        dataloader = train_data.into_dataloader(copied_model._batch_size)
+
+        loss_fn = nn.MSELoss()
+
+        optimizer = torch.optim.SGD(copied_model._model.parameters(), lr=0.05)
+
+        loss_values = []
+        accuracies = []
+        for _ in range(epoch_size):
+            tmp_loss = []
+            tmp_accuracies = []
```

Comment:
Consider using a running loss/accuracy which is the current sum of the loss/accuracy in the epoch. You can get the average by dividing through the total number of batches*batch_size